File: bench/elixir/nqueens.exs
defmodule Q do
  def safe?(_c, [], _d), do: true

  def safe?(c, [p | rest], d) do
    cond do
      p == c -> false
      p - c == d -> false
      p - c == -d -> false
      true -> safe?(c, rest, d + 1)
    end
  end

  def solve(row, n, _placed) when row == n, do: 1

  def solve(row, n, placed) do
    Enum.reduce(0..(n - 1), 0, fn c, acc ->
      if safe?(c, placed, 1), do: acc + solve(row + 1, n, [c | placed]), else: acc
    end)
  end
end

n = String.to_integer(System.get_env("BENCH_N") || "10")
IO.puts(Q.solve(0, n, []))

File: bench/erlang/nqueens.erl
#!/usr/bin/env escript
%%! -mode(compile)
main(_) -> io:format("~w~n", [solve(0, bench_n(10), [])]).
bench_n(D) -> case os:getenv("BENCH_N") of false -> D; V -> list_to_integer(V) end.
safe(_C, [], _D) -> true;
safe(C, [P | Rest], D) ->
    if P == C -> false; P - C == D -> false; P - C == -D -> false; true -> safe(C, Rest, D+1) end.
solve(Row, N, _P) when Row == N -> 1;
solve(Row, N, Placed) ->
    lists:foldl(fun(C, Acc) ->
        case safe(C, Placed, 1) of true -> Acc + solve(Row+1, N, [C|Placed]); false -> Acc end
    end, 0, lists:seq(0, N-1)).
